extensions/wikia/Privacy/RemoveUserDataOnWikiTask.php

```diff
@@ -220,28 +251,52 @@ public function removeUserDataOnCurrentWiki( $auditLogId, $userId, $username, $o
 		$results[] = $this->removeIpFromRecentChanges( $userId );
 		$results[] = $this->removeWatchlist( $userId );
 
-		$userDbKey = Title::newFromText( $username )->getDBkey();
+		$userDbKey = Title::newFromText( $user->getName() )->getDBkey();
 		$results[] = $this->removeUserPages( $userDbKey );
 		$results[] = $this->removeUserPagesFromRecentChanges( $userDbKey );
 		$results[] = $this->removeActionLogs( $userDbKey );
-		if ( !empty( $oldUsername ) ) {
-			$oldUserDbKey = Title::newFromText( $oldUsername )->getDBkey();
+		if ( !empty( $oldUserId ) ) {
+			$oldUserDbKey = Title::newFromText( $oldUser->getName() )->getDBkey();
 			$results[] = $this->removeUserPages( $oldUserDbKey );
 			$results[] = $this->removeUserPagesFromRecentChanges( $oldUserDbKey );
 			$results[] = $this->removeActionLogs( $oldUserDbKey );
 		}
 
-		RemovalAuditLog::markTaskAsFinished(
-			$auditLogId,
-			$wgCityId,
-			array_reduce( $results, function( $acc, $res ) { return $acc && $res; }, true ) );
+		RemovalAuditLog::markTaskAsFinished( $auditLogId, $wgCityId,
+			array_reduce( $results, function ( $acc, $res ) { return $acc && $res; }, true ) );
 
 		$this->info( "Removed user data from wiki" );
+
+		if ( RemovalAuditLog::allWikiDataWasRemoved( $auditLogId ) ) {
```

Comment:
Would you mind adding context here, as to why are we waiting for all data to be removed?

Reply:
Comment would be just fine.